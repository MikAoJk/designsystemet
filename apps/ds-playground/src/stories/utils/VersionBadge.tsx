import { ReactNode } from 'react';

interface VersionBadgeProps {
  packageName: string;
}

export const VersionBadge = ({ packageName }: VersionBadgeProps): ReactNode => {
  let color = 'green';

  let version = process.env[`@skatteetaten/ds-${packageName}`] ?? '..';
  if (process.env['NODE_ENV'] === 'development') {
    color = 'red';
    version = 'developing';
  }
  if (process.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    color = 'blue';
  }

  const normalizedPackageName = packageName.replaceAll('-', '--');
  const normalizedVersion = version.replaceAll('-', '--').replaceAll('_', '__');

  return (
    <img
      src={`https://img.shields.io/badge/%40skatteetaten%2Fds--${normalizedPackageName}-${normalizedVersion}-${color}?logo=npm`}
      alt={`@skatteetaten/ds-${packageName}@${version}`}
    />
  );
};
