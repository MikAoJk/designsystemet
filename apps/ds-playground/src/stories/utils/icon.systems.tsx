import { SystemSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const SystemSVGPaths = findSVGPathElements(SystemSVGPathsAndIcons);
export const SystemIconLabels = makeLabelsFromSVGPathElements(SystemSVGPaths);
