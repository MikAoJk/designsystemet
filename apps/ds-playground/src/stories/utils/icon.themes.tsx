import { ThemeSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const ThemeSVGPaths = findSVGPathElements(ThemeSVGPathsAndIcons);
export const ThemeIconLabels = makeLabelsFromSVGPathElements(ThemeSVGPaths);
