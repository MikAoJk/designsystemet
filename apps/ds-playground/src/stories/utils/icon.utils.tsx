import {
  SystemSVGPathsAndIcons,
  ThemeSVGPathsAndIcons,
} from '@skatteetaten/ds-icons';

export type SVGPathAndIconType =
  | typeof SystemSVGPathsAndIcons
  | typeof ThemeSVGPathsAndIcons;
export type SVGPathAndIconKey = keyof SVGPathAndIconType;

export const makeLabelsFromSVGPathElements = function (
  SVGPathElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, string> {
  const labels: Record<string, string> = {};
  Object.keys(SVGPathElements).forEach((name: string) => {
    labels[name] = name.replace('SVGpath', '');
  });

  return labels;
};

export const findSVGPathElements = function (
  SVGPathAndIconElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, SVGPathAndIconType[SVGPathAndIconKey]> {
  const elements: Record<string, SVGPathAndIconType[SVGPathAndIconKey]> = {};
  Object.keys(SVGPathAndIconElements).forEach((elementName: string) => {
    if (elementName.endsWith('SVGpath')) {
      elements[elementName] =
        SVGPathAndIconElements[elementName as SVGPathAndIconKey];
    }
  });

  return elements;
};
