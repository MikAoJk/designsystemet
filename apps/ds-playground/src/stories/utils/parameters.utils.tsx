import { ArgsTable, Stories, PRIMARY_STORY } from '@storybook/addon-docs';
import type { Parameters } from '@storybook/csf';

type DesignParameters = {
  name: string;
  url: string;
};

export const getParameters = function (
  designParameters: DesignParameters[]
): Parameters {
  return {
    docs: {
      page: (): JSX.Element => {
        return (
          <>
            <Stories title={''} />
            <ArgsTable story={PRIMARY_STORY} />
          </>
        );
      },
    },
    design: designParameters.map((designParameter: DesignParameters) => {
      return {
        name: designParameter.name,
        type: 'figma',
        url: designParameter.url,
      };
    }),
    backgrounds: {
      values: [
        { name: 'Svart', value: '#000' },
        { name: 'Hvit', value: '#fff' },
      ],
    },
  };
};
