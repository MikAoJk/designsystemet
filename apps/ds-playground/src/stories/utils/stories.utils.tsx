export const exampleParameters = {
  docs: {
    source: {
      type: 'code',
      language: 'tsx',
      transform: (source: string): string => {
        return source
          .replace('render: (_args): JSX.Element', 'const Examples = ()')
          .replace(
            'render: function Render(_args): JSX.Element ',
            'const Examples = () => '
          )
          .replace('satisfies Story', '');
      },
    },
  },
  /* _args i en story må finnes slik at props i fanen Controls skjules og er prefikset med underscore for å unngå eslint klaging */
  controls: {
    exclude: /.*/,
  },
};
