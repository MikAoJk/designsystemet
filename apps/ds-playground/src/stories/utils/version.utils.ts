export const getVersion = (
  packageName: string
): {
  major: string;
  minor: string;
  patch: string;
  postfix?: string;
  style?: Record<string, string>;
} => {
  const ver = process.env[`@skatteetaten/${packageName}`] ?? '..';
  const [major, minor, patch] = ver.split('.');
  const majorWithPackageName = packageName ? `${packageName}@${major}` : major;

  const style = {
    color: 'var(--semantic-page-foreground)',
    background: 'var(--semantic-success-background)',
    'font-weight': 'var(--font-weight-bold)',
    'font-size': '13px',
  };

  if (process.env['NODE_ENV'] === 'development') {
    return {
      major: majorWithPackageName,
      minor,
      patch,
      postfix: 'develop',
      style: {
        ...style,
        background: 'var(--semantic-danger-background)',
      },
    };
  }
  if (process.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    return {
      major: majorWithPackageName,
      minor,
      patch,
      postfix: process.env['STORYBOOK_BUILD_VERSION'],
      style: {
        ...style,
        background: 'var(--semantic-warning-background)',
      },
    };
  }

  return {
    major: majorWithPackageName,
    minor,
    patch,

    style: {
      ...style,
    },
  };
};
