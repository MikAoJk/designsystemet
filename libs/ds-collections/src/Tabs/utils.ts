export const valueRegex = /^[\w:.-_]*$/;
