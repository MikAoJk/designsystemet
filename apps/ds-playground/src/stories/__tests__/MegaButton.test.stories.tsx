import { useState } from 'react';

import {
  MegaButton,
  MegaButtonComponentCommonProps,
  MegaButtonDiscriminatedProp,
} from '@skatteetaten/ds-buttons';
import { expect } from '@storybook/jest';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { userEvent, waitFor, within } from '@storybook/testing-library';

import { wrapper } from './testUtils/storybook.testing.utils';

const defaultMegaButtonText = 'Klikk her';

const verifyAttribute =
  (attribute: string, expectedValue: string) =>
  async ({ canvasElement }: { canvasElement: HTMLElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const megaButton = canvas.getByRole('button');
    await expect(megaButton).toBeInTheDocument();
    await expect(megaButton).toHaveAttribute(attribute, expectedValue);
  };

export default {
  component: MegaButton,
  title: 'Tester/MegaButton',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: { table: { disable: true } },
    isExternal: { table: { disable: true } },
    // HTML
    accessKey: {
      table: { disable: true },
      control: 'text',
    },
    disabled: { table: { disable: true } },
    href: { table: { disable: true } },
    type: { table: { disable: true } },
    // Aria
    ariaDescribedby: { table: { disable: true } },
    // Events
    onBlur: { table: { disable: true } },
    onClick: { table: { disable: true } },
    onFocus: { table: { disable: true } },
  },
} as ComponentMeta<typeof MegaButton>;

const Template: ComponentStory<typeof MegaButton> = (args) => (
  <div className={'noTransition'} data-test-block>
    {/* eslint-disable-next-line testing-library/no-node-access */}
    <MegaButton {...args}>{args.children}</MegaButton>
  </div>
);

const defaultArgs: MegaButtonComponentCommonProps = {
  children: defaultMegaButtonText,
};

// Når MegaButton har en ref, så får dom button elementet ref forwarded
export const WithRef = Template.bind({});
WithRef.storyName = 'With Ref (FA1)';
WithRef.args = {
  ...defaultArgs,
  ref: (instance: HTMLButtonElement | null): void => {
    if (instance) {
      instance.id = 'dummyIdForwardedFromRef';
    }
  },
};
WithRef.argTypes = {
  ...WithRef.argTypes,
  ref: { table: { disable: false } },
};
WithRef.parameters = {
  imageSnapshot: { disable: true },
};
WithRef.play = verifyAttribute('id', 'dummyIdForwardedFromRef');

// Når MegaButton har en id, så har button-element id
// Når MegaButton har en custom CSS, så vises custom stil
// Når MegaButton har en lang, så har button-element lang
// Når MegaButton har dataTestid, så har button-elementet data-testid satt
export const WithAttributes = Template.bind({});
WithAttributes.storyName = 'With Attributes(FA2-5)';
WithAttributes.args = {
  ...defaultArgs,
  id: 'htmlId',
  className: 'dummyClassname',
  lang: 'nb',
  'data-testid': '123ID',
};
WithAttributes.argTypes = {
  ...WithAttributes.argTypes,
  id: { table: { disable: false } },
  className: { table: { disable: false } },
  lang: { table: { disable: false } },
  'data-testid': { table: { disable: false } },
};
WithAttributes.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const megaButton = canvas.getByRole('button');
  await expect(megaButton).toHaveClass('dummyClassname');
  await expect(megaButton).toHaveAttribute('id', 'htmlId');
  await expect(megaButton).toHaveAttribute('lang', 'nb');
  await expect(megaButton).toHaveAttribute('data-testid', '123ID');
};

// Når MegaButton instansieres, så får den riktige default-verdier og rendrer riktig i ulike tilstander
export const Defaults = Template.bind({});
Defaults.storyName = 'Defaults (A1, B2)';
Defaults.args = {
  ...defaultArgs,
};
Defaults.argTypes = {
  ...Defaults.argTypes,
  children: { table: { disable: false } },
};
Defaults.parameters = {
  imageSnapshot: {
    focus: `${wrapper} > button`,
    hover: `${wrapper} > button`,
    click: `${wrapper} > button`,
  },
};
Defaults.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const megaButton = canvas.getByText(defaultMegaButtonText);
  await expect(megaButton).toBeInTheDocument();
  await expect(megaButton).toHaveAttribute('type', 'button');
};

// Når MegaButton har en veldig lang tekst så skal det brekke over flere linjer
export const WithLongText = Template.bind({});
WithLongText.storyName = 'With Long Text (A2)';
WithLongText.args = {
  ...defaultArgs,
  children: 'Denne knappen har en veldig lang tekst. Så lang at den må brekke.',
};
WithLongText.argTypes = {
  ...WithLongText.argTypes,
  children: { table: { disable: false } },
};

// Når MegaButton har en veldig lang tekst uten breaking space så skal det brekke over flere linjer
export const WithLongTextBreaking = Template.bind({});
WithLongTextBreaking.storyName = 'With Long Text Breaking (A1, A2)';
WithLongTextBreaking.args = {
  ...defaultArgs,
  children: 'Denneknappenharenveldiglangtekst.Sålangatdenmåbrekke.',
};
WithLongTextBreaking.argTypes = {
  ...WithLongTextBreaking.argTypes,
  children: { table: { disable: false } },
};

// Når MegaButton har isExternal, så vises riktig ikon med tilhørende tekst. Tester også for de icon props som er relevant for saken med systemIcon som er brukt.
export const WithExternalIcon = Template.bind({});
WithExternalIcon.storyName = 'With External Icon (A4, B5, B7)';
WithExternalIcon.args = {
  ...defaultArgs,
  isExternal: true,
};
WithExternalIcon.argTypes = {
  ...WithExternalIcon.argTypes,
  isExternal: { table: { disable: false } },
};
WithExternalIcon.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);

  const megaButton = canvas.getByRole('button');
  // eslint-disable-next-line testing-library/no-node-access
  const svg = megaButton.querySelector('svg');
  await expect(svg).toHaveAttribute('aria-label', 'Til et annet nettsted');
  await expect(svg).toHaveAttribute('viewBox', '0 0 24 24');
  await expect(megaButton).toBeInTheDocument();
};

// Når MegaButton har en veldig lang tekst så skal teksten brytes over flere linjer
// og når det er et ikon skal ikonet plasseres løpende etter teksten
export const WithLongTextAndExternalIcon = Template.bind({});
WithLongTextAndExternalIcon.storyName = 'With Long Text and External Icon (A2)';
WithLongTextAndExternalIcon.args = {
  ...defaultArgs,
  isExternal: true,
  children:
    'Denne knappen har en veldig lang tekst. Icon skal da plasseres løpende etter tekster på siste linje',
};
WithLongTextAndExternalIcon.argTypes = {
  ...WithLongTextAndExternalIcon.argTypes,
  children: { table: { disable: false } },
  isExternal: { table: { disable: false } },
};

export const Disabled = Template.bind({});
Disabled.storyName = 'Disabled (B6)';
const discriminatedProps: MegaButtonDiscriminatedProp = {
  href: undefined,
  disabled: true,
};
Disabled.args = {
  ...defaultArgs,
  ...discriminatedProps,
};
Disabled.argTypes = {
  ...Disabled.argTypes,
  disabled: { table: { disable: false } },
};
Disabled.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  expect(canvas.getByText(defaultMegaButtonText)).toBeDisabled();
};

// Når MegaButton har prop type, så har button-elementet type satt
export const WithType = Template.bind({});
WithType.storyName = 'With Type (B2)';
WithType.args = {
  ...defaultArgs,
  type: 'submit',
};
WithType.argTypes = {
  ...WithType.argTypes,
  type: { table: { disable: false } },
};
WithType.parameters = {
  imageSnapshot: { disable: true },
};
WithType.play = verifyAttribute('type', 'submit');

// Når MegaButton har aria attributer, så har button element aria-* satt
export const WithAriaDescribedby = Template.bind({});
WithAriaDescribedby.storyName = 'With AriaDescribedby (B1)';
WithAriaDescribedby.args = {
  ...defaultArgs,
  ariaDescribedby: 'testid1234',
};
WithAriaDescribedby.argTypes = {
  ...WithAriaDescribedby.argTypes,
  ariaDescribedby: { table: { disable: false } },
};
WithAriaDescribedby.parameters = {
  imageSnapshot: { disable: true },
};
WithAriaDescribedby.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const megaButton = canvas.getByRole('button');
  await expect(megaButton).toBeInTheDocument();
  await expect(megaButton).toHaveAttribute('aria-describedby', 'testid1234');
};

// Når MegaButton har en accessKey, så settes den som forventet
export const WithAccesskey = Template.bind({});
WithAccesskey.storyName = 'With Accesskey (B4)';
WithAccesskey.args = {
  children: defaultMegaButtonText,
  accessKey: 'j',
};
WithAccesskey.argTypes = {
  ...WithAccesskey.argTypes,
  accessKey: { table: { disable: false } },
};
WithAccesskey.parameters = {
  imageSnapshot: { disable: true },
};
WithAccesskey.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const megaButton = canvas.getByRole('button');
  await expect(megaButton).toBeInTheDocument();
  await expect(megaButton).toHaveAttribute('accessKey', 'j');
};

// Når MegaButton har en href, så rendres den som en a
export const AsLink = Template.bind({});
AsLink.storyName = 'As Link (B3)';
AsLink.args = {
  ...defaultArgs,
  href: 'https://www.skatteetaten.no',
};
AsLink.argTypes = {
  ...AsLink.argTypes,
  href: { table: { disable: false } },
};
AsLink.parameters = {
  imageSnapshot: {
    focus: `${wrapper} > a`,
    hover: `${wrapper} > a`,
  },
};

// Når MegaButton har en href med tom streng, så rendres den som en a
export const AsLinkEmptyString = Template.bind({});
AsLinkEmptyString.storyName = 'As Link with empty href (B3)';
AsLinkEmptyString.args = {
  ...defaultArgs,
  href: '',
};
AsLinkEmptyString.argTypes = {
  ...AsLinkEmptyString.argTypes,
  href: { table: { disable: false } },
};
AsLinkEmptyString.parameters = {
  imageSnapshot: { disable: true },
};

// Når MegaButton har en href og er eksternlink, så rendres den som en a og det vises eksternlink-ikon
export const AsLinkExternal = Template.bind({});
AsLinkExternal.storyName = 'As Link External (B3, A4)';
AsLinkExternal.args = {
  ...defaultArgs,
  href: 'https://www.skatteetaten.no',
  isExternal: true,
};
AsLinkExternal.argTypes = {
  ...AsLinkExternal.argTypes,
  href: { table: { disable: false } },
  isExternal: { table: { disable: false } },
};
AsLinkExternal.parameters = {
  imageSnapshot: {
    focus: `${wrapper} > a`,
    hover: `${wrapper} > a`,
  },
};
AsLinkExternal.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const megaButton = canvas.getByText('Klikk her');
  await expect(megaButton).toBeInTheDocument();
  await expect(megaButton.tagName).toBe('A');
  await expect(megaButton).toHaveAttribute('role', 'button');
};

// Når brukeren setter focus, blurrer, eller klikker på knappen, så kalles riktig eventHandler
// Eventhandlere endrer tesksten på knappen
const EventHandlersTemplate: ComponentStory<typeof MegaButton> = (args) => {
  const [buttonText, setButtonText] = useState('bruk knapp for å teste events');
  return (
    <div className={'noTransition'} data-test-block>
      <MegaButton
        {...args}
        onFocus={(
          event: React.FocusEvent<HTMLButtonElement | HTMLAnchorElement>
        ): void => {
          setButtonText('Knapp har fått fokus');
          args.onFocus && args.onFocus(event);
        }}
        onBlur={(
          event: React.FocusEvent<HTMLButtonElement | HTMLAnchorElement>
        ): void => {
          setButtonText('Knapp har blitt blurret');
          args.onBlur && args.onBlur(event);
        }}
        onClick={(
          event: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement>
        ): void => {
          setButtonText('Knapp har blitt klikket på');
          args.onClick && args.onClick(event);
        }}
      >
        {buttonText}
      </MegaButton>
    </div>
  );
};
export const WithEventHandlers = EventHandlersTemplate.bind({});
WithEventHandlers.storyName = 'With EventHandlers (A2 delvis)';
WithEventHandlers.args = {
  ...defaultArgs,
};
WithEventHandlers.argTypes = {
  ...WithEventHandlers.argTypes,
};
WithEventHandlers.parameters = {
  imageSnapshot: { disable: true },
};
WithEventHandlers.play = async ({ args, canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const megaButton = canvas.getByText('bruk knapp for å teste events');
  await expect(megaButton).toBeInTheDocument();
  await megaButton.focus();
  await waitFor(() => expect(args.onFocus).toHaveBeenCalled());
  await userEvent.tab();
  await waitFor(() => expect(args.onBlur).toHaveBeenCalled());
  await userEvent.click(megaButton);
  await waitFor(() => expect(args.onClick).toHaveBeenCalled());
};
