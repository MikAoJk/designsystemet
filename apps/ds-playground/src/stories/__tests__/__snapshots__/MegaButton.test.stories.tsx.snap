// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/MegaButton AsLink smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="MegaButton_button__njALO "
     href="https://www.skatteetaten.no"
     role="button"
  >
    Klikk her
  </a>
</div>
`;

exports[`Tester/MegaButton AsLinkEmptyString smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="MegaButton_button__njALO "
     href
  >
    Klikk her
  </a>
</div>
`;

exports[`Tester/MegaButton AsLinkExternal play-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="MegaButton_button__njALO "
     href="https://www.skatteetaten.no"
     role="button"
  >
    Klikk her
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz MegaButton_icon__KSvBN"
         viewbox="0 0 24 24"
         focusable="false"
         aria-label="Til et annet nettsted"
         aria-hidden="false"
         role="img"
    >
      <path d="M14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7Zm5 16H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7h-2v7Z">
      </path>
    </svg>
  </a>
</div>
`;

exports[`Tester/MegaButton Defaults play-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <button class="MegaButton_button__njALO "
          type="button"
  >
    Klikk her
  </button>
</div>
`;

exports[`Tester/MegaButton WithAccesskey play-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <button class="MegaButton_button__njALO "
          accesskey="j"
          type="button"
  >
    Klikk her
  </button>
</div>
`;

exports[`Tester/MegaButton WithAriaDescribedby play-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <button class="MegaButton_button__njALO "
          type="button"
          aria-describedby="testid1234"
  >
    Klikk her
  </button>
</div>
`;

exports[`Tester/MegaButton WithAttributes play-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <button id="htmlId"
          class="MegaButton_button__njALO dummyClassname"
          lang="nb"
          data-testid="123ID"
          type="button"
  >
    Klikk her
  </button>
</div>
`;

exports[`Tester/MegaButton WithDisabled play-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <button class="MegaButton_button__njALO "
          disabled
          type="button"
  >
    Klikk her
  </button>
</div>
`;

exports[`Tester/MegaButton WithEventHandlers play-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <button class="MegaButton_button__njALO "
          type="button"
  >
    Knapp har blitt klikket på
  </button>
</div>
`;

exports[`Tester/MegaButton WithExternalIcon play-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <button class="MegaButton_button__njALO "
          type="button"
  >
    Klikk her
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz MegaButton_icon__KSvBN"
         viewbox="0 0 24 24"
         focusable="false"
         aria-label="Til et annet nettsted"
         aria-hidden="false"
         role="img"
    >
      <path d="M14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7Zm5 16H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7h-2v7Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/MegaButton WithLongText smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <button class="MegaButton_button__njALO "
          type="button"
  >
    Denne knappen har en veldig lang tekst. Så lang at den må brekke.
  </button>
</div>
`;

exports[`Tester/MegaButton WithLongTextAndBreaking smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <button class="MegaButton_button__njALO "
          type="button"
  >
    Denneknappenharenveldiglangtekst.Sålangatdenmåbrekke.
  </button>
</div>
`;

exports[`Tester/MegaButton WithLongTextAndExternalIcon smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <button class="MegaButton_button__njALO "
          type="button"
  >
    Denne knappen har en veldig lang tekst. Icon skal da plasseres løpende etter tekster på siste linje
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz MegaButton_icon__KSvBN"
         viewbox="0 0 24 24"
         focusable="false"
         aria-label="Til et annet nettsted"
         aria-hidden="false"
         role="img"
    >
      <path d="M14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7Zm5 16H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7h-2v7Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/MegaButton WithRef play-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <button class="MegaButton_button__njALO "
          type="button"
          id="dummyIdForwardedFromRef"
  >
    Klikk her
  </button>
</div>
`;

exports[`Tester/MegaButton WithType play-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <button class="MegaButton_button__njALO "
          type="submit"
  >
    Klikk her
  </button>
</div>
`;
